src/transformers/models/mt5/modeling_mt5.py

```diff
@@ -1451,6 +1481,17 @@ def forward(
         >>> outputs = model(input_ids=input_ids, decoder_input_ids=decoder_input_ids)
         >>> last_hidden_states = outputs.last_hidden_state
         ```"""
+        # Copied from models.bart.modeling_bart.BartModel.forward different to other models, MT5 automatically creates
+        # decoder_input_ids from input_ids if no decoder_input_ids are provided
+        if decoder_input_ids is None and decoder_inputs_embeds is None:
+            if input_ids is None:
+                raise ValueError(
+                    "If no `decoder_input_ids` or `decoder_inputs_embeds` are "
+                    "passed, `input_ids` cannot be `None`. Please pass either "
+                    "`input_ids` or `decoder_input_ids` or `decoder_inputs_embeds`."
+                )
+            decoder_input_ids = self._shift_right(input_ids)
```

Comment:
Not sure this is also valid for `MT5` and the `t5` family?

Reply:
Yeah I'm not a 100% on this, but I found I needed this to be able to use the `run_glue.py` script to train a text classification model. I did get good results with that so it does seem to work. Without this though the script exits in an error about not finding any decoder input ids.

Reply:
Okay I gotcha. Just to clarify on next steps then. Should I reverse the changes in this commit https://github.com/huggingface/transformers/pull/24726/commits/d36122bb9e6f6aacee827142f31979844bb504cd where I switched to using the `base_model` as @sgugger suggested? 

I'm asking since I'm not sure it's possible to only have these changes in the `ForSequenceClassification`'s forward and also use the `base_model`.

Reply:
And indeed it is. I also checked locally with `run_glue.py` and training works as expected.